packages/thumbnail-grid-block/src/components/Thumbnail/Item.tsx

```diff
@@ -27,28 +35,53 @@ export const Item = forwardRef<HTMLDivElement, ThumbnailItemProps>(
             draggableProps = {},
             setActivatorNodeRef,
             isDragging = false,
-            setUploadedId,
-            openAssetChooser,
-            openFileDialog,
-            updateItemWith,
-            onRemoveAsset,
+            updateItem,
+            onRemoveItem,
             showGrabHandle,
+            showDeleteButton,
             thumbnailStyles,
             isLoading,
-            onFilesDrop,
+            onFilesSelected,
+            onFilesUploaded,
+            onAssetsSelected,
             replaceWithPlaceholder = false,
         },
         ref,
     ) => {
         const [showAltTextMenu, setShowAltTextMenu] = useState(false);
         const { id, title, description, altText } = item;
         const [localAltText, setLocalAltText] = useState<string | undefined>(altText);
+        const [openFileDialog, { selectedFiles }] = useFileInput({ accept: 'image/*', multiple: true });
+        const [uploadFile, { results: uploadResults, doneAll }] = useAssetUpload();
 
-        const onOpenFileDialog = () => {
-            setUploadedId(id);
-            openFileDialog();
+        const onOpenAssetChooser = () => {
+            appBridge.dispatch(
+                openAssetChooser({
+                    multiSelection: true,
+                    selectedValueId: image?.id,
+                    objectTypes: [AssetChooserObjectType.ImageVideo],
+                    extensions: FileExtensionSets.Images,
+                }),
+            );
+            const unsusbcribe = appBridge.subscribe('assetsChosen', ({ assets }) => {
+                onAssetsSelected(assets, id);
+                appBridge.dispatch(closeAssetChooser());
+                unsusbcribe();
+            });
         };
-        const onAssetChooserClick = () => openAssetChooser(id);
+
+        useEffect(() => {
+            if (selectedFiles) {
+                onFilesSelected(selectedFiles, id);
+                uploadFile(selectedFiles);
+            }
+        }, [selectedFiles]);
```

Comment:
```suggestion
        }, [selectedFiles, onFilesSelected, uploadFile, id]);
```

Reply:
Those dependencies change too often and can result in faulty functioning. To be in line with other uses of this hook I added a line to disable the warning.

Reply:
Thought they wouldn't change so its okay.